### PR TITLE
Bump version to v1beta1

### PR DESCRIPTION
**Pull request by nirarg**
Alighn cluster-api versions to cluster-api:
* Increase Go to 1.17
* Change api version to v1beta1

### CODE REVIEW

**Comment by nirarg** [comment]
@agradouski  FYI

**Review by davidvossel — COMMENTED** [review]
can you update the `kubevirtci` tools to download clusterctl that is compatible with v1beta1

Also, if you haven't already, try running `hack/functest.sh` to see if this works with the upcoming CI job.

**Comment by agradouski** [comment]
@nirarg awesome! thanks for this PR. I will give it a test on our internal infrastructure and let you know how it goes

**Comment by nirarg** [comment]
> can you update the `kubevirtci` tools to download clusterctl that is compatible with v1beta1
> 
> Also, if you haven't already, try running `hack/functest.sh` to see if this works with the upcoming CI job.

@davidvossel done, see https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/pull/33/commits/ef6d28f724403125376ccac0c1a0956cc74e4341

**Review by davidvossel — COMMENTED** [review]
looks good

I ran the `hack/functest.sh` locally and everything passed for me. My environment was on golang 1.16, so building was a challenge. To get this to work for me, I replaced the current Dockerfile with a multi-stage dockerfile that builds using 1.17. 

```
# syntax = docker/dockerfile:1-experimental

FROM golang:1.17 AS base
ENV CGO_ENABLED=0
WORKDIR /src
COPY ./ /src
RUN go mod download

FROM base AS build
RUN mkdir /out
RUN --mount=target=. \
  --mount=type=cache,target=/root/.cache/go-build \
  CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o /out/manager

FROM gcr.io/distroless/static
COPY --from=build /out/manager /usr/bin/
ENTRYPOINT /usr/bin/manager  
```

Any thoughts on if we should create a build container of sorts, so these types of bumps to the golang version don't impact our local environments?

**Comment by davidvossel** [comment]
I made a dockerized build script #42 that allows us to decouple our local environments from whatever golang version we want to build capk with.

File: api/v1beta1/kubevirtmachinetemplate_types.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KubevirtMachineTemplateSpec defines the desired state of KubevirtMachineTemplate.
+type KubevirtMachineTemplateSpec struct {
```
Comment (davidvossel) [inline api/v1beta1/kubevirtmachinetemplate_types.go:24]: After giving this some more thought, I'm unsure if we should be incrementing the capk object's to beta, but instead ensure that we work properly with the capi v1beta1 apis.

Our capk apis are still alpha, and will certainly change (see #11 ) so moving to beta is probably premature. That hopefully doesn't inhibit us adopting the capi v1beta1 api though. It seems like these two things should be orthogonal to one another.

**Comment by davidvossel** [comment]
/hold

let's discuss this in the community call tomorrow (tuesday 14th)

**Comment by nirarg** [comment]
I'm closing this PR
Will open a different PR for changing back to v1alpha1

**Comment by nirarg** [comment]
Opened new PR: https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/pull/50

**Pull request closed without merging**